python/src/aiconfig/default_parsers/openai.py

```diff
@@ -74,40 +50,81 @@ def serialize(
+                break
+
+        # Get the global settings for the model
+        model_name = conversation_data["model"] if "model" in conversation_data else self.id()
+
+        model_metadata = ai_config.generate_model_metadata(conversation_data, model_name)
+        # Remove messages array from model metadata. Handled separately
+        model_metadata.settings.pop("messages", None)
+
+        # Add in system prompt
+        if system_prompt:
+            model_metadata.settings["system_prompt"] = system_prompt
+
+        if not prompt_name:
+            prompt_name = "prompt"
+
+        i = 0
+        while i < len(conversation_data["messages"]):
+            messsage = conversation_data["messages"][i]
+            current_message_role = messsage["role"]
+            if current_message_role == "user" or current_message_role == "function":
+                # Serialize User message as a prompt and save the assistant response as an output
+                assistant_response = None
+                if i + 1 < len(conversation_data["messages"]):
+                    next_message = conversation_data["messages"][i + 1]
+                    if next_message["role"] == "assistant":
+                        assistant_response = next_message
+                        i += 1
+                new_prompt_name = (
+                    f"{prompt_name}_{len(serialized_prompts) + len(ai_config.prompts) +1}"
+                )
 
-            override_settings = {key: prompt_model_metadata[key] for key in override_keys}
+                input_to_use = (
+                    messsage["content"]
+                    if current_message_role == "user"
+                    else PromptInput(**messsage)
+                )
 
-            if override_settings:
-                model_metadata = {"name": model_name, "settings": override_settings}
-            else:
-                model_metadata = model_name
-        else:
-            model_metadata = {"name": model_name, "settings": prompt_model_metadata}
-
-        return Prompt(
-            name=prompt_name,
-            input=prompt_input,
-            metadata=PromptMetadata(model=PromptMetadata(**model_metadata)),
-            parameters=parameters,
-            **kwargs
-        )
+                user_prompt = Prompt(
+                    name=new_prompt_name,
+                    input=input_to_use,
+                    metadata=PromptMetadata(
+                        **{
+                            "model": copy.deepcopy(model_metadata),
+                            "parameters": parameters,
+                            "remember_chat_context": True,
+                        }
+                    ),
+                    outputs=[
+                        ExecuteResult(
+                            output_type="execute_result",
+                            execution_count=None,
+                            data=assistant_response,
+                            metadata={},
```

Comment:
`metadata=None`

Reply:
Metadata is defined to be of type `Dict`, pydantic will enforce this and not allow None type unless explicitly defined with `None` or `Optional`.

Will make a new diff on top of the stack to refactor to allow None as a default value. Will also update other types as per @jonathanlastmileai 's previous suggestion regarding setting defaults as `None` and handling from there.